seedsource_core/django/seedsource/management/commands/calculate_zone_transfers.py

```diff
@@ -7,268 +7,369 @@
 from trefoil.utilities.color import Color
 from trefoil.utilities.window import Window
 
-from ..utils import ZoneConfig
-
-VARIABLES = (
-    'MAT', 'MWMT', 'MCMT', 'TD', 'MAP', 'MSP', 'AHM', 'SHM', 'DD_0', 'DD5', 'FFP', 'PAS', 'EMT', 'EXT', 'Eref', 'CMD'
+from seedsource_core.django.seedsource.management.constants import VARIABLES
+from seedsource_core.django.seedsource.management.utils import (
+    get_region_for_zone,
+    calculate_pixel_area,
 )
+from seedsource_core.django.seedsource.management.dataset import (
+    ElevationDataset,
+    ClimateDatasets,
+)
+from seedsource_core.django.seedsource.management.zoneconfig import ZoneConfig
+
+
+PERIODS = ("1961_1990", "1981_2010")
 
 
 class Command(BaseCommand):
-    help = 'Calculates default variable transfer limits for each available seed zone'
+    help = "Calculates default variable transfer limits for each available seed zone"
 
     def __init__(self, *args, **kwargs):
         self.transfers_by_source = {}
 
         super().__init__(*args, **kwargs)
 
     def add_arguments(self, parser):
-        parser.add_argument('source_name', nargs='?', type=str, default='all', help='Accepts zone name or all')
-
-    def _get_subsets(self, elevation, data, coords: SpatialCoordinateVariables, bbox):
-        """ Returns subsets of elevation, data, and coords, clipped to the given bounds """
 
-        x_slice = slice(*coords.x.indices_for_range(bbox.xmin, bbox.xmax))
-        y_slice = slice(*coords.y.indices_for_range(bbox.ymin, bbox.ymax))
+        parser.add_argument(
+            "--zones",
+            dest="zoneset",
+            default=None,
+            help="Comma delimited list of zones sets to analyze. (default is to analyze all available zone sets)",
+        )
 
-        if x_slice.stop - x_slice.start < 2:
-            x_slice = slice(x_slice.start, x_slice.start + 2)
-        if y_slice.stop - y_slice.start < 2:
-            y_slice = slice(y_slice.start, y_slice.start + 2)
+        parser.add_argument(
+            "--variables",
+            dest="variables",
+            default=None,
+            help="Comma delimited list of variables analyze. (default is to analyze all available variables)",
+        )
 
-        return elevation[y_slice, x_slice], data[y_slice, x_slice], coords.slice_by_window(Window(y_slice, x_slice))
+        parser.add_argument(
+            "--clear",
+            dest="clear",
+            action="store_true",
+            default=False,
+            help="If True, will automatically delete all transfer limits before calculating new ones",
+        )
 
-    def _write_limit(self, variable, time_period, zone, masked_data, low=None, high=None, label=None):
-        min_value = numpy.nanmin(masked_data)
-        max_value = numpy.nanmax(masked_data)
+    def _write_limit(
+        self, variable, time_period, zone, masked_data, band, elevation_service
+    ):
+        min_value = masked_data.min()
+        max_value = masked_data.max()
```

Comment:
`.min()` and `.max()` don't work correctly if the array contains any `nan` values.

Reply:
The data are extracted within the mask prior to calling this function, so there should never be `nan` values here.  I've added a docstring to make this more clear, since `masked_data` isn't super clear in isolation.